std/zig/tokenizer.zig

```diff
@@ -591,11 +585,11 @@ pub const Tokenizer = struct {
                 State.C => switch (c) {
                     '\\' => {
                         state = State.Backslash;
-                        result.id = Token.Id{ .MultilineStringLiteralLine = Token.StrLitKind.C };
+                        result.id = Token.Id.MultilineStringLiteralLine;
```

Comment:
Before merging, I'd like to understand why destroying this information didn't result in any test failures.

Reply:
Is it because I removed the switch statement?
https://github.com/ziglang/zig/pull/1584/commits/f808bc46f8061e68f2f1ba3689108f30d8cdddc6#diff-240205850d93cadbcbe8eefbba911c75L1350

Reply:
Here's why: https://github.com/ziglang/zig/blob/c5509a07cacfea16b4fb3c8d552a33083b713697/std/zig/parse_string_literal.zig#L22

I think this PR is fine. #265 will happen before the self hosted compiler needs to distinguish between the 2 kinds of string literal kinds.